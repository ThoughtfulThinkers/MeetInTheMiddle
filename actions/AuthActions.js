import firebase from 'firebase';
import { Actions } from 'react-native-router-flux';
import {
  EMAIL_CHANGED,
  FETCH_GEOLOCATION_BY_FULL_ADDRESS_SUCCESS,
  LOGIN_USER_SUCCESS,
  LOGIN_USER_FAIL,
  LOGIN_USER,
  LOAD_AUTHENTICATED_USER_STATE_SUCCESS,
  SET_LOGIN_STATUS,
  PASSWORD_CHANGED,
} from './types';

import { googlePlacesConfig, GOOGLE_GEO_API_KEY } from '../envConfig';

/*****************************************************************
  Changes to form Input fields
  TODO: move to UserActions.js & Refactor that to FormActions.js
*****************************************************************/

export const emailChanged = (text) => {
  return {
    type: EMAIL_CHANGED,
    payload: text
  };
};

export const passwordChanged = (text) => {
  return {
    type: PASSWORD_CHANGED,
    payload: text
  };
};

/*****************************************************************
    Access Actions
    setLoginStatus is used to cofirm a user's login status
    by app components.  This is especially useful when
    accessing a screen that requires a user be logged in.
*****************************************************************/
export const setLoginStatus = status => {
  return {
    type: SET_LOGIN_STATUS,
    payload: status
  };
};

export const loginUser = ({ email, password }) => {
  return (dispatch) => {
    dispatch({ type: LOGIN_USER });

    firebase.auth().signInWithEmailAndPassword(email, password)
      .then(user => {
        loginUserSuccess(dispatch, user);
        dispatch(loadAuthenticatedUserState());
        return user;
      })
      .then(() => Actions.meetups({ type: 'reset' }))
      .catch((error) => {
        console.log(error);
      });
  };
};

const loginUserSuccess = (dispatch, user) => {
  dispatch({
    type: LOGIN_USER_SUCCESS,
    payload: user
  });
};

export const logoutUser = () => dispatch => {
  firebase.auth().signOut()
    .then(() => Actions.meetups({ type: 'reset' }))
    .catch(error => console.log('Sign Out Error ', error));
};

const loginUserFail = (dispatch) => {
  dispatch({ type: LOGIN_USER_FAIL });
  Actions.login({ type: 'reset' });
};

/*****************************************************************
  Manage User
  loadAuthenticatedUserState is used both in the physical login
  process and to keep the user state between app restarts
*****************************************************************/

export const loadAuthenticatedUserState = () => dispatch => {
  const { currentUser } = firebase.auth();
  if (!currentUser) {
    dispatch({ type: LOAD_AUTHENTICATED_USER_STATE_SUCCESS, payload: {} });
  }
  firebase.database().ref(`/users/${currentUser.uid}`)
    .on('value', (snapshot) => {
      dispatch({ type: LOAD_AUTHENTICATED_USER_STATE_SUCCESS, payload: snapshot.val() });
    });
};

/*****************************************************************
    Create New User Account

    Creating a new user account is a three step process:
    1. Create the user account in FB
    2. Get the user's latitude & logitude from Google based on
       their prefered location address
    2. Create a user profile (e.g. name, location) in FB
*****************************************************************/

export const createNewUserAccount = userProfileData => dispatch => {
  const { email, password } = userProfileData;
  firebase.auth().createUserWithEmailAndPassword(email, password)
    .then(user => {
      let newUserData = userProfileData;
      newUserData = { ...newUserData, uid: user.uid };
      createNewUserProfile(dispatch, newUserData);
    })
    .catch(() => loginUserFail(dispatch));
};

const createNewUserProfile = (dispatch, newUserData) => {
    const { currentUser } = firebase.auth();
    const { street, city, state, zipcode, firstName, lastName } = newUserData;
    const fullAddress = `${street},${city},${state}`;
    const url = `https://maps.googleapis.com/maps/api/geocode/json?address=${fullAddress}&key=${GOOGLE_GEO_API_KEY}`;
    fetch(url)
      .then(response => response.json())
      .then(data => {
        const lat = data.results[0].geometry.location.lat;
        const lon = data.results[0].geometry.location.lng;
        const location = { lat, lon, street, city, state, zipcode };
        const userData = {
          uid: currentUser.uid,
          firstName,
          lastName,
          location
        };
        dispatch({ type: FETCH_GEOLOCATION_BY_FULL_ADDRESS_SUCCESS, payload: location });
        dispatch(() => setNewUser(dispatch, userData));
      })
      .catch(error => console.log('fetchGeoLocationByFullAddress error: ', error.message));
  // };
};

export const setNewUser = (dispatch, userData) => {
  const { currentUser } = firebase.auth();
  firebase.database().ref(`/users/${currentUser.uid}`)
    .set(userData)
    .then(() => Actions.meetups({ type: 'reset' }))
    .catch(error => console.log(error));
};

/*****************************************************************
    Update User Account
    This is a 2 step process
    1. Get the GeoLocation (lat/lon) from Google based on user's
       profile location
    2. Update FB user data
*****************************************************************/

export const updateUser = data => {
  return dispatch => {
    const { currentUser } = firebase.auth();
    const { street, city, state, zipcode, firstName, lastName, image } = data;
    const fullAddress = `${street},${city},${state}`;
    const url = `https://maps.googleapis.com/maps/api/geocode/json?address=${fullAddress}&key=${GOOGLE_GEO_API_KEY}`;
    fetch(url)
    .then(response => response.json())
      .then(data => {
        // console.log('location: ', data.results[0].geometry.location);
        let userData = {};
        // const latLon = data.results[0].geometry.location;
        // let location = { lat: latLon.lat, lon: latLon.lng };
        const lat = data.results[0].geometry.location.lat;
        const lon = data.results[0].geometry.location.lng;
        let location = { lat, lon };
        if (state.length > 0) location = { ...location, street, city, state, zipcode }; // state required at minimum
        userData = { ...userData, location };
        // Firebase doesn't allow empty documents on an update
        if (firstName.length > 0) userData = { ...userData, firstName };
        if (lastName.length > 0) userData = { ...userData, lastName };
        if (image.length > 0) userData = { ...userData, image };
        dispatch({ type: FETCH_GEOLOCATION_BY_FULL_ADDRESS_SUCCESS, payload: location });
        firebase.database().ref(`/users/${currentUser.uid}`)
          .update(userData)
          .then(response => Actions.meetups({ type: 'reset' }))
          .catch(error => console.log('updateUser Error: ', error));
      })
      .catch(error => console.log('fetchGeoLocationByFullAddress error: ', error.message));
  };
};

// TODO: Change email and password
// export const updateUserEmail

// TODO: Send password reset email
export const emailPasswordRest = () => dispatch => {
  
};

// TODO: Authenticate Email address

// TODO: Add displayName and photoURL w/i user's auth profile
