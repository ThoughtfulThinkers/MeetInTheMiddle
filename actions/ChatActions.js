import firebase from 'firebase';
import { Actions } from 'react-native-router-flux';

import {
  FETCH_CHAT_MESSAGES_SUCCESS
} from './types';


export const fetchChatMessages = () => {
  // TODO: if not current user, send to login page
  const { currentUser } = firebase.auth();
  return dispatch => {
    firebase.database().ref('messages')
      .on('value', snapshot => {
        dispatch({ type: FETCH_CHAT_MESSAGES_SUCCESS, payload: snapshot.val() });
      });
  };
};
