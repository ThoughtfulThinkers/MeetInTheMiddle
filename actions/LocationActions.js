import { Actions } from 'react-native-router-flux';
import 'whatwg-fetch';
import { googlePlacesConfig } from '../envConfig';

import {
  FETCH_GEOLOCATION_BY_FULL_ADDRESS_SUCCESS
} from './types';

/**************************************************
  Google
**************************************************/
// const GOOGLE_PLACES_API = googlePlacesConfig.apiKey;
const GOOGLE_API = 'AIzaSyDzk0eKI5tnKWkSORpDTL32iZ15QjxQxeg';
export const fetchGeoLocationByFullAddress = (street, city, state) => dispatch => {
  const fullAddress = `${street},${city},${state}`;
  const url = `https://maps.googleapis.com/maps/api/geocode/json?address=${fullAddress}&key=${GOOGLE_API}`;
  fetch(url)
    .then(response => response.json())
    .then(data => {
      // console.log('location: ', data.results[0].geometry.location);
      const location = data.results[0].geometry.location;
      dispatch({ type: FETCH_GEOLOCATION_BY_FULL_ADDRESS_SUCCESS, payload: location });
    })
    .catch(error => console.log('fetchGeoLocationByFullAddress error: ', error));
};
