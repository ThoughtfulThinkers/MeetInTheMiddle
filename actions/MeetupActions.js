import firebase from 'firebase';
import { Actions } from 'react-native-router-flux';

import {
  FETCH_MEETUPS_SUCCESS,
  FETCH_USER_MEETUPS_SUCCESS,
  MEETUP_CREATE_SUCCESS
} from './types';

export const meetupsFetch = (city) => {
  const { currentUser } = firebase.auth();

  return (dispatch) => {
    const ref = firebase.database().ref('/meetups');
    ref.orderByChild('city').equalTo(city).on('value', (snapshot) => {
      // console.log(snapshot.val());
      dispatch({ type: FETCH_MEETUPS_SUCCESS, payload: snapshot.val() });
    });
  };
};

export const userMeetupsFetch = () => {
  const { currentUser } = firebase.auth();

  return (dispatch) => {
    firebase.database().ref(`/users/${currentUser}/meetups`).on('child_added', (snapshot) => {
      const meetups = snapshot.val().map((meetup) => {
        // console.log('Testing key, ', meetup.key);
        return firebase.database().ref('/meetups').orderByKey().equalTo(meetup.key);
      });
      Promise.all(meetups)
      .then((data) => {
        // console.log('User fetch meetups success: ', data);
        dispatch({ type: FETCH_USER_MEETUPS_SUCCESS, payload: data });
      })
      .catch((err) => {
        console.log(err);
      });
    });
  };
};

export const createMeetup = (meetup) => {
  const { currentUser } = firebase.auth();

  return (dispatch) => {
  firebase.database().ref('/meetups')
    .push(meetup)
    .then(({ key }) => {
      // console.log('created meetup: ', key);
      dispatch({ type: MEETUP_CREATE_SUCCESS });
      Actions.meetup({ type: 'reset', id: key });
    });
  };
};
