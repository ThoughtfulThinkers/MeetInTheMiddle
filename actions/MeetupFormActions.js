import firebase from 'firebase';
import { Actions } from 'react-native-router-flux';
import {
  MEETUP_CHANGED,
  ADD_MEETUP,
  ADD_MEETUP_SUCCESS,
  SET_CURRENT_MEETUP,
  EDIT_MEETUP,
  EDIT_MEETUP_SUCCESS
} from './types';

export const meetupChange = (prop, value) => {
  return {
    type: MEETUP_CHANGED,
    prop,
    value
  };
};

export const addMeetup = (meetupDetails) => {
  return (dispatch) => {
  dispatch({ type: ADD_MEETUP });
  //const { currentUser } = firebase.auth();
  const meetup = {
    ...meetupDetails,
    chat: {},
    vote: {},
    users: {},
    location: '',
    status: 'created'
  };
  firebase.database().ref('/meetups')
    .push(meetup)
    .then(({ key }) => {
      dispatch({ type: ADD_MEETUP_SUCCESS });
      Actions.meetup({ type: 'reset', meetup });
    })
    .catch((err) => console.log(err));
  };
};

export const setCurrentMeetup = (meetup) => {
  return {
    type: SET_CURRENT_MEETUP,
    meetup
  };
};

export const meetupEdit = (meetup) => {
  return (dispatch) => {
  dispatch({ type: EDIT_MEETUP });
  //const { currentUser } = firebase.auth();
  const { name, description, start, end, state, venue, voteStart, voteEnd } = meetup;
  const updates = {};
  updates['/name'] = name;
  updates['/description'] = description;
  updates['/start'] = start;
  updates['/end'] = end;
  updates['/state'] = state;
  updates['/venue'] = venue;
  updates['/voteStart'] = voteStart;
  updates['/voteEnd'] = voteEnd;
  updates['/status'] = 'created';

  firebase.database().ref(`/meetups/${meetup.uid}`)
    .update(updates)
    .then(() => {
      dispatch({ type: EDIT_MEETUP_SUCCESS });
      Actions.meetups({ type: 'reset' });
    })
    .catch((err) => console.log(err));
  };
};

export const changeStatus = (meetup, status) => {
  console.log('change status to ', status);
  return (dispatch) => {
    firebase.database().ref(`/meetups/${meetup.uid}/status`)
      .set(status)
      .then(() => {
        console.log('set meetup');

        dispatch({
            type: MEETUP_CHANGED,
            prop: 'status',
            value: status });
      })
      .catch((err) => console.log(err));
    };
  };
