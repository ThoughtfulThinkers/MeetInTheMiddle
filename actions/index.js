export * from './AuthActions';
export * from './ChatActions';
export * from './MeetupActions';
export * from './UserActions';
export * from './LocationActions';
export * from './FilterActions';
export * from './MeetupFormActions';
export * from './RsvpActions';
export * from './MapActions';
