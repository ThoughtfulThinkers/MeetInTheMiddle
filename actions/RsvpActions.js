import firebase from 'firebase';
import { Actions } from 'react-native-router-flux';
import { googlePlacesConfig } from '../envConfig';
import {
        CHANGE_RSVP,
        SET_RSVP,
        SET_RSVP_SUCCESS,
        DELETE_RSVP_SUCCESS,
        } from '../actions/types';
import { userMeetupsFetch } from './MeetupActions';
import {
          getUser,
          setGuest,
          setMeetup,
          updateGuest,
          updateMeetup,
          removeGuest,
          removeMeetup,
          updateStatus,
        } from './firebase-functions/RsvpActions';

const { apiKey } = googlePlacesConfig;

export const changeRSVP = (lat, lon) => {
  return dispatch => {
    const search = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${lon}&key=${apiKey}`;
    return fetch(search)
    .then(response => response.json())
    .then((data) => {
      const address = data.results[0].formatted_address;
      dispatch({
        type: CHANGE_RSVP,
        lat,
        lon,
        address
      });
    })
    .catch(err => console.log(err));
  };
};

export const setRsvp = (lat, lon, meetupId, users, name) => {
  return (dispatch) => {
    dispatch({ type: SET_RSVP });

    const currentUser = getUser();
    if (!currentUser) {
      Actions.login();
      return;
    }

    const userId = currentUser.uid;
    const guest = {
      uid: userId,
      name,
      lat,
      lon
    };

    return setGuest(meetupId, userId, guest)
      .then(() => {
        const currentMeetup = { lat, lon, uid: meetupId };
        return setMeetup(userId, currentMeetup)
        .then(() => {
          dispatch({ type: SET_RSVP_SUCCESS });
          Actions.meetups({ type: 'reset' });
        });
      })
      .catch((err) => console.log(err));
  };
};

export const editRsvp = (lat, lon, meetupId, users, name) => {
  return (dispatch) => {
    dispatch({ type: SET_RSVP });

    const currentUser = getUser();
    if (!currentUser) {
      Actions.login();
      return;
    }

    const userId = currentUser.uid;
    const guest = {
      uid: userId,
      name,
      lat,
      lon
    };

    const updates = {};
    updates[`/${userId}`] = guest;

    return updateGuest(meetupId, updates)
    .then(() => {
      const currentMeetup = { lat, lon, uid: meetupId };
      return updateMeetup(userId, currentMeetup)
      .then(() => {
        dispatch({ type: SET_RSVP_SUCCESS });
        Actions.meetups({ type: 'reset' });
      });
    })
    .catch((err) => console.log(err));
  };
};

export const deleteRsvp = (meetup) => {
  return (dispatch) => {
    dispatch({ type: SET_RSVP });

    const currentUser = getUser();
    if (!currentUser) {
      Actions.login();
      return;
    }
    const meetupId = meetup.uid;
    const userId = currentUser.uid;

    return removeGuest(meetupId, userId)
      .then(() => {
        return removeMeetup(meetupId, userId)
          .then(() => {
            return updateStatus(meetupId)
            .then(() => {
              const newUsers = meetup.users;
              delete newUsers[userId];
              const newMeetup = meetup;
              newMeetup.users = newUsers;
              dispatch({ type: DELETE_RSVP_SUCCESS, meetup: newMeetup, id: meetupId });
              dispatch(userMeetupsFetch());
              Actions.meetups({ type: 'reset' });
            })
            .catch(err => console.log(err));
          })
          .catch(err => console.log(err));
        })
    .catch((err) => console.log(err));
  };
};
