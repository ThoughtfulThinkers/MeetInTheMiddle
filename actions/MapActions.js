import 'whatwg-fetch';
import firebase from 'firebase';
import { Actions } from 'react-native-router-flux';
import { changeStatus } from './MeetupFormActions';
import { SET_CURRENT_MEETUP, SET_VOTE, MEETUP_CHANGED } from './types';
import { foursquareConfig, googlePlacesConfig } from '../envConfig';

const { ID, SECRET } = foursquareConfig;
const { apiKey } = googlePlacesConfig;
const venueIds = [{ name: 'Arts & Entertainment', id: '4d4b7104d754a06370d81259' },
{ name: 'College & University', id: '4d4b7105d754a06372d81259' },
{ name: 'Event', id: '4d4b7105d754a06373d81259' },
{ name: 'Food', id: '4d4b7105d754a06374d81259' },
{ name: 'Coffee Shop', id: '4bf58dd8d48988d1e0931735' },
{ name: 'Fast Food Restaurant', id: '4bf58dd8d48988d16e941735' },
{ name: 'Nightlife Spot', id: '4d4b7105d754a06376d81259' },
{ name: 'Bar', id: '4bf58dd8d48988d116941735' },
{ name: 'Outdoors & Recreation', id: '4d4b7105d754a06377d81259' },
{ name: 'Park', id: '4bf58dd8d48988d163941735' },
{ name: 'Professional & Other Places', id: '4d4b7105d754a06375d81259' },
{ name: 'Community Center', id: '52e81612bcbc57f1066b7a34' },
{ name: 'Spiritual Center', id: '4bf58dd8d48988d131941735' },
{ name: 'Shop & Service', id: '4d4b7105d754a06378d81259' },
{ name: 'Travel & Transport', id: '4d4b7105d754a06379d81259' }];

export const createVoting = (lat, lon, meetup) => {
  const venueType = meetup.venue.id;
  return dispatch => {
    const search = `https://api.foursquare.com/v2/venues/explore?ll=${lat},${lon}&client_id=${ID}&client_secret=${SECRET}&sortByDistance=true&section=${venueType}&limit=10&v=20170201&m=foursquare`;
    fetch(search)
    .then(response => response.json())
    .then(data => {
      let venues;
      if (!data.response.groups || data.response.groups.length === 0 || data.response.groups[0].items.length === 0) {
        const search2 = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${lon}&key=${apiKey}`;
        fetch(search2)
        .then(response => response.json())
        .then((data2) => {
          let formattedAddress;
          if (!data2 || !data2.results || !data2.results[0] || !data2.results[0].formatted_address) {
            formattedAddress = 'Invalid Location';
          } else {
            formattedAddress = data2.results[0].formatted_address;
          }
          const location = {
            formattedAddress: [formattedAddress],
            lat,
            lon,
            name: formattedAddress,
            votes: 0
          };
          dispatch(changeLocation(location, meetup.uid));
          dispatch(changeStatus(meetup, 'set'));
        })
        .catch(err => console.log(err));
      } else {
        venues = {};
        data.response.groups[0].items.forEach(item => {
          const { venue } = item;
          const { name, location, id } = venue;
          const { formattedAddress, lng } = location;
          venues[id] = { name, formattedAddress, lat: location.lat, lon: lng, votes: 0 };
        });
      }
      const newMeetup = { ...meetup, status: 'voting', venues };
      firebase.database().ref(`/meetups/${meetup.uid}`)
      .set(newMeetup)
      .then(() => {
        dispatch({ type: SET_CURRENT_MEETUP, meetup: newMeetup });
        Actions.meetup({ type: 'refresh' });
      })
      .catch(err => console.log('create venues error', err));
    })
    .catch(err => console.log(err));
  };
};

export const setVote = (meetupId, venueId, venueVote) => {
  return dispatch => {
  const { currentUser } = firebase.auth();
  firebase.database().ref(`/users/${currentUser.uid}/meetups/${meetupId}/vote`)
    .set(venueId)
    .then(() => {
      firebase.database().ref(`/meetups/${meetupId}/venues/${venueId}/votes`)
      .set(venueVote)
      .then(() => {
        dispatch({ type: SET_VOTE, venueId, vote: venueVote });
        Actions.meetup({ type: 'refresh' });
      });
    })
    .catch(err => console.log(err));
  };
};

export const voteChange = (meetupId, venueId, venueVote) => {
  return dispatch => {
      const { currentUser } = firebase.auth();

      //get old vote location
      let userPreviousVote;
      firebase.database().ref(`/users/${currentUser.uid}/meetups/${meetupId}/vote`)
        .on('value', (snapshot) => {
          userPreviousVote = snapshot.val();
      });

      //get old venue vote count
      let oldVenueCount;
      firebase.database().ref(`/meetups/${meetupId}/venues/${userPreviousVote}/votes`)
        .on('value', (snapshot) => {
          oldVenueCount = snapshot.val();
      });

      //lower old vote count, raise new vote count, replace user venue id
      oldVenueCount -= 1;
      const updates = {};
      updates[`/users/${currentUser.uid}/meetups/${meetupId}/vote`] = venueId;
      updates[`/meetups/${meetupId}/venues/${userPreviousVote}/votes`] = oldVenueCount;
      updates[`/meetups/${meetupId}/venues/${venueId}/votes`] = venueVote;

      firebase.database().ref().update(updates).then(() => {
        dispatch({ type: SET_VOTE, venueId, vote: venueVote });
        Actions.meetup({ type: 'refresh' });
      });
  };
};

export const changeLocation = (location, meetupId) => {
  return (dispatch) => {
    firebase.database().ref(`/meetups/${meetupId}/location`)
      .set(location)
      .then(() => {
        dispatch({
            type: MEETUP_CHANGED,
            prop: 'location',
            value: location });
            Actions.meetup({ type: 'refresh' });
      })
      .catch((err) => console.log(err));
    };
};
