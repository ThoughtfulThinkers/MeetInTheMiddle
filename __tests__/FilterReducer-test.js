 import FilterReducer from '../reducers/FilterReducer';

 import * as actions from '../actions';

describe('filterReducer', () => {
 it('undefined', () => {
   expect(FilterReducer(undefined, { type: 'null' })).toMatchSnapshot();
 });

 it('setText', () => {
   expect(FilterReducer(undefined, actions.setText('Penn'))).toMatchSnapshot();
 });

 it('setLocation', () => {
   expect(FilterReducer(undefined, actions.setLocation('Pennsylvania'))).toMatchSnapshot();
 });
});
