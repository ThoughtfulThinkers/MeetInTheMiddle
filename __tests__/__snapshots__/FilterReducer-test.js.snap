exports[`filterReducer setLocation 1`] = `
Object {
  "location": "Pennsylvania",
  "text": "",
}
`;

exports[`filterReducer setText 1`] = `
Object {
  "location": "New York",
  "text": "Penn",
}
`;

exports[`filterReducer undefined 1`] = `
Object {
  "location": "New York",
  "text": "",
}
`;
